### ISSUE TITLE
Sporadic failure: ContainerKeyIndexTests testRecovery

### ISSUE THREAD

**Issue by shiveshr** [pravega/pravega#4202]
**Problem description**
Encountered failure in test on travis. 
```
io.pravega.segmentstore.server.tables.ContainerKeyIndexTests > testRecovery FAILED
    java.lang.AssertionError: Expected getBucketOffsets() to block.
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at io.pravega.segmentstore.server.tables.ContainerKeyIndexTests.testRecovery(ContainerKeyIndexTests.java:469)
```

**Problem location**
ContainerKeyIndexTests.java

**Suggestions for an improvement**
Fix the issue

**Comment by shiveshr** [comment]
another failure in the same test but a different error this time:
```
io.pravega.segmentstore.server.tables.ContainerKeyIndexTests > testRecovery FAILED

    java.lang.AssertionError: Not expecting anything to be unblocked at this point

        at org.junit.Assert.fail(Assert.java:88)

        at org.junit.Assert.assertTrue(Assert.java:41)

        at org.junit.Assert.assertFalse(Assert.java:64)

        at io.pravega.segmentstore.server.tables.ContainerKeyIndexTests.testRecovery(ContainerKeyIndexTests.java:485)```

**Comment by andreipaduroiu** [comment]
duplicate of #4193
